### PR TITLE
Tutorial island branch

### PR DESCRIPTION
**Pull request by CurtisGreen**
Added tutorial island and addressed issues with autoclickers that came with it. Autoclickers are now destroyed when leaving levels and recreated when opening a new level rather than paused and unpaused. Everything functions as expected, just need to get a better background picture in the future.

### CODE REVIEW

File: source/constants.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const CONSTANTS = {
         MENU: "MENU",
         CC: "CC",
         MAP: "MAP",
+        TUTORIALISLAND: "TUTORIALISLAND",
```
Comment (EthanVieira) [inline source/constants.js]: TUTORIAL_ISLAND

File: source/healthBar.js [hunk 2]
```diff
@@ -5,10 +5,12 @@ export class HealthBar {
 	x = 0;
 	y = 0;
 	height = 0;
+    width = 0;
```
Comment (EthanVieira) [inline source/healthBar.js:8]: tab

File: source/healthBar.js [hunk 3]
```diff
@@ -5,10 +5,12 @@ export class HealthBar {
...
 	constructor(scene, x, y, maxHealth){
 		// Offset coordinates based on width/height of health bar
-		this.x = x - 4 - (5*maxHealth/12);
-		this.height = 12;
+        this.width = maxHealth + 20;
```
Comment (EthanVieira) [inline source/healthBar.js:11]: tab

File: source/healthBar.js [hunk 4]
```diff
@@ -19,20 +21,23 @@ export class HealthBar {
 		this.healthBar = scene.add.graphics();
 		this.healthBar.setDepth(3);
         this.healthBar.fillStyle(0x000000); // Border (black)
-        this.healthBar.fillRect(this.x, this.y, this.maxHealth+4, this.height+4);
+        this.healthBar.fillRect(this.x, this.y, this.width+4, this.height+4);
```
Comment (EthanVieira) [inline source/healthBar.js:24]: tab

File: source/level.js [hunk 5]
```diff
@@ -7,29 +7,46 @@ export class Level extends Phaser.Scene{
...
+            questCompleted: false,
+            enemiesKilled: 0
         }
     };
     // Autoclickers
     autoClickers = [];
     autoClickDps = 0;
     // Enemy
-	enemySettings; // From constructor
+	enemySettings = { // From constructor
```
Comment (EthanVieira) [inline source/level.js]: weird tab here, just look through file again for weird tabbing - it happens elsewhere

File: source/scenes/lumbridge.js [hunk 6]
```diff
@@ -11,7 +11,7 @@ export class Lumbridge extends Level{
             enemy: {
                 name: 'cow', 
                 path: 'source/assets/sprites/cow.png',
-                maxHealth: 20,
+                maxHealth: 10,
```
Comment (EthanVieira) [inline source/scenes/lumbridge.js]: 8

File: source/scenes/tutorialIsland.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { CONSTANTS } from "../constants.js";
+import { Level } from "../level.js";
+
+export class TutorialIsland extends Level{
+    constructor() {
+        super({
+            key: CONSTANTS.SCENES.TUTORIALISLAND,
+            killQuest: 10,
+            background: { name: 'tutorialIsland', path: 'source/assets/Tutorial_Island_Mine.png' },
+            minimap: { name: 'tutorialIslandMap', path: 'source/assets/TutorialIslandMap.png'},
```
Comment (EthanVieira) [inline source/scenes/tutorialIsland.js:10]: we should prob start being consistent with naming assets -> _ or no _?

**Review by EthanVieira — CHANGES_REQUESTED** [review]

File: source/healthBar.js (hunk 2 above)
Comment (CurtisGreen) [reply source/healthBar.js:8]: Newest commit converted tabs to spaces (like most ides do automatically) so git will parse it properly. Should be fine from now on

File: source/scenes/lumbridge.js (hunk 6 above)
Comment (CurtisGreen) [reply source/scenes/lumbridge.js]: K

File: source/scenes/tutorialIsland.js (hunk 7 above)
Comment (CurtisGreen) [reply source/scenes/tutorialIsland.js:10]: I'm partial to no_ but idc as long as we're consistent

Comment (EthanVieira) [reply source/scenes/tutorialIsland.js:10]: Added a task on trello to deal with this

File: source/constants.js (hunk 1 above)
Comment (CurtisGreen) [reply source/constants.js]: k

**Review by EthanVieira — APPROVED** [review]

**Pull request merged**